components/Ripple/src/MDCRippleTouchController.h

```diff
@@ -0,0 +1,184 @@
+// Copyright 2019-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import <UIKit/UIKit.h>
+#import "MDCRippleView.h"
+
+@protocol MDCRippleTouchControllerDelegate;
+
+/**
+ Provides the current state of the ripple. The ripple is either in its normal state, or in the
+ selected state where the ripple remains spread on the view.
+
+ - MDCRippleStateNormal: The ripple is in its normal state.
+ - MDCRippleStateSelected: The ripple is in the selected state.
+ */
+typedef NS_ENUM(NSInteger, MDCRippleState) {
+  MDCRippleStateNormal = 0,
+  MDCRippleStateSelected,
+};
+
+/**
+ The MDCRippleTouchController is a convenience controller that adds all the needed touch tracking
+ and logic to provide the correct ripple effect based on the user interacting with the view the
+ ripple is added to.
+ */
+@interface MDCRippleTouchController : NSObject <UIGestureRecognizerDelegate>
+
+/**
+ A weak reference to the view that the ripple is added to, and that responds to the user
+ events.
+ */
+@property(nonatomic, weak, readonly, nullable) UIView *view;
+
+/**
+ The ripple view added to the view.
+ */
+@property(nonatomic, strong, readonly, nonnull) MDCRippleView *rippleView;
+
+/** Delegate to extend the behavior of the touch control. */
+
+/**
+ A delegate to extend the behavior of the touch controller.
+ */
+@property(nonatomic, weak, nullable) id<MDCRippleTouchControllerDelegate> delegate;
+
+/** Gesture recognizer used to bind touch events to ripple. */
+
+/**
+ The gesture recognizer used to bind the touch events to the ripple.
+ */
+@property(nonatomic, strong, readonly, nonnull) UILongPressGestureRecognizer *gestureRecognizer;
+
+/**
+ The selection gesture recognizer used to bind the touch events related to selection to the ripple.
+ */
+@property(nonatomic, strong, readonly, nullable)
+    UILongPressGestureRecognizer *selectionGestureRecognizer;
+
+/**
+ This BOOL tells the touch controller if to allow selection and all the logic and visuals
```

Comment:
_Tip:_ This BOOL tells the touch controller ~~if~~ to allow ..., the _if_ sounds oddly worded to me, deferring to you but just want to make sure this is worded how you intended.